src/jit/lowerxarch.cpp

```diff
@@ -1110,6 +1116,23 @@ void Lowering::TreeNodeInfoInitCall(GenTreeCall* call)
         // computed into a register.
         if (!call->IsFastTailCall())
         {
+#ifdef _TARGET_X86_
+            // On x86, we need to generate a very specific pattern for indirect VSD calls:
+            //
+            //    3-byte nop
+            //    call dword ptr [eax]
+            //
+            // Where EAX is also used as an argument to the stub dispatch helper. Make
+            // sure that the call target address is computed into EAX in this case.
+            if (((call->gtFlags & GTF_CALL_VIRT_KIND_MASK) == GTF_CALL_VIRT_STUB) && (call->gtCallType == CT_INDIRECT))
+            {
+                assert(ctrlExpr->isIndir());
+
+                ctrlExpr->gtGetOp1()->gtLsraInfo.setDstCandidates(l, RBM_EAX);
```

Comment:
>               ctrlExpr->gtGetOp1()->gtLsraInfo.setDstCandidates(l, RBM_EAX); [](start = 1, length = 77)

This makes me realize the same bug is present on amd64, because TreeNodeInfo::Initialize() is setting DstCandidates, in case of VSD calls.
Probably this went unnoticed is because GT_CALL node immediately follows ctrlExpr in execution order and LSRA stress modes might have not caused a spill there.
Feel free to open an issue for tracking amd64 issue, if you don't want to club it with this PR.